numpy/core/src/umath/ufunc_object.c

```diff
@@ -2868,32 +2868,42 @@ PyUFunc_Reduce(PyUFuncObject *ufunc, PyArrayObject *arr, PyArrayObject *out,
     if (identity == NULL) {
         return NULL;
     }
+
+    /* Get the initializer */
+    if (initializer == NULL) {
+        initializer = identity;
```

Comment:
I think the `identity` should only be gotten here, inside the loop. After all, it is not used if an `initializer` is passed on, so one might as well skip getting it.

Reply:
We need `reorderable`, which we get from the same function. I did that at first but was having failures.

Reply:
Ah, that makes sense. That's what one gets for coming to the discussion late...